cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -572,7 +621,8 @@ KafkaReconciler kafkaReconciler(List<KafkaNodePool> nodePools, KafkaCluster kafk
                     config,
                     supplier,
                     pfa,
-                    vertx
+                    vertx,
+                    kafkaMetadataStateManager
```

Comment:
Again, I do not understand why does this need the manager and not just the state.

Reply:
Even in this case the manager instance is used for checking if migration is in progress via the `checkMigrationInProgress` called during the reconciliation within the `updateKafkaMetadataMigrationState`.

Reply:
I think it is worth separating it. Among other things, it will IMHO significantly simplify the testing if you can just pass the state and not the whole Manager. You can also easily test the node deletion separately etc.

Reply:
But the Kafka reconciler is also using the manager instance method `computeNextMetadataState` to compute the next FSM state ad the end of the reconciliation as result of the current state (at the beginning of the reconcile) + the annotation applied and the actions taken. It's actually computing the transition and it's stored in the `Kafka` CR status.